internal/controller/numaresourcesscheduler_controller_test.go

```diff
@@ -631,6 +633,167 @@ var _ = ginkgo.Describe("Test NUMAResourcesScheduler Reconcile", func() {
 			gomega.Expect(*dp.Spec.Replicas).To(gomega.Equal(int32(numOfMasters)), "number of replicas is different than number of control-planes nodes; want=%d got=%d", numOfMasters, *dp.Spec.Replicas)
 		})
 	})
+
+	ginkgo.Context("with kubelet PodResourcesAPI listing active pods by default", func() {
+		var nrs *nropv1.NUMAResourcesScheduler
+		var reconciler *NUMAResourcesSchedulerReconciler
+		numOfMasters := 3
+
+		ginkgo.When("kubelet fix is enabled", func() {
+			fixedVersion, _ := platform.ParseVersion(activePodsResourcesSupportSince)
+
+			ginkgo.DescribeTable("should configure by default the informerMode to the expected when field is not set", func(reconcilerPlatInfo PlatformInfo) {
+				var err error
+				nrs = testobjs.NewNUMAResourcesScheduler("numaresourcesscheduler", "some/url:latest", testSchedulerName, 11*time.Second)
+				initObjects := []runtime.Object{nrs}
+				initObjects = append(initObjects, fakeNodes(numOfMasters, 3)...)
+				reconciler, err = NewFakeNUMAResourcesSchedulerReconciler(initObjects...)
+				gomega.Expect(err).ToNot(gomega.HaveOccurred())
+
+				reconciler.PlatformInfo = reconcilerPlatInfo
+
+				key := client.ObjectKeyFromObject(nrs)
+				_, err = reconciler.Reconcile(context.TODO(), reconcile.Request{NamespacedName: key})
+				gomega.Expect(err).ToNot(gomega.HaveOccurred())
+				expectedInformer := depmanifests.CacheInformerDedicated
+				if reconcilerPlatInfo.Platform == platform.OpenShift || reconcilerPlatInfo.Platform == platform.HyperShift {
+					expectedInformer = depmanifests.CacheInformerShared
+				}
```

Comment:
this should be a test parameter so it's trivial to verified. If we want to determine implicitly it should also be a function of version, which is basically rewriting the production code logic. Having it explicit bypasses this problem.

Reply:
I see your point and makes sense indeed, fixed by passing the expected value.